src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -217,9 +213,33 @@ export async function generateRawTransaction(args: {
     ? Promise.resolve({ gas_estimate: options.gasUnitPrice })
     : getGasPriceEstimation({ aptosConfig });
 
-  const [{ sequence_number: sequenceNumber }, { chain_id: chainId }, { gas_estimate: gasEstimate }] = await Promise.all(
-    [getSequenceNumber, getChainId, getGasUnitPrice],
-  );
+  const [{ chain_id: chainId }, { gas_estimate: gasEstimate }] = await Promise.all([getChainId, getGasUnitPrice]);
+
+  const getSequenceNumber =
+    options?.accountSequenceNumber !== undefined
+      ? Promise.resolve({ sequence_number: options.accountSequenceNumber })
+      : getInfo({ aptosConfig, accountAddress: sender });
+
+  let sequenceNumber: string | AnyNumber;
+
+  /**
+   * Check if is sponsored transaction to honor AIP-52
+   * {@link https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-52.md}
+   */
+  if (feePayerAddress === AccountAddress.ZERO.toString()) {
```

Comment:
I think fee payer address doesn't necessarily have to be a string?

I think you need to resolve the fee payer address prior to the comparison.

e.g.
```
AccountAddress.from(feePayerAddress) === AccountAddress.ZERO
```

Reply:
updated